addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScannerUnitTest.java

```diff
@@ -0,0 +1,301 @@
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInCookieOfResponse() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET / HTTP/1.1");
+        String jso = Base64.getEncoder().encodeToString(createJso());
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Set-Cookie: CRUNCHY=" + jso + "\r\n");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInRawBodyOfResponse()
+            throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET / HTTP/1.1");
+        byte[] jso = createJso();
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Content-Type: application/octet-stream\r\n"
+                        + "Content-Disposition: attachment; filename=\"jso.bin\"\r\n"
+                        + "Content-Length: "
+                        + jso.length
+                        + "\r\n");
+        msg.setResponseBody(jso);
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInBase64BodyOfResponse()
+            throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET / HTTP/1.1");
+        String jso = Base64.getEncoder().encodeToString(createJso());
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Content-Type: application/octet-stream\r\n"
+                        + "Content-Disposition: attachment; filename=\"jso.bin\"\r\n"
+                        + "Content-Length: "
+                        + jso.length()
+                        + "\r\n");
+        msg.setResponseBody(jso);
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenUriEncodedJsoMagicBytesAreDetectedInRequestParameterOfRequest()
+            throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET /some_action?q=" + URI_ENCODED_JSO + "&p=&m HTTP/1.1");
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenBase64JsoMagicBytesAreDetectedInRequestParameterOfResponse()
```

Comment:
Was this added on purpose? It's still testing the request.

Reply:
Yes, it should be named about the request, not about the response